spring-ai-test/src/main/java/org/springframework/ai/test/vectorstore/BaseVectorStoreTests.java

```diff
@@ -110,7 +110,8 @@ protected void deleteById() {
 			assertThat(results.get(0).getId()).isEqualTo(documents.get(2).getId());
 			Map<String, Object> metadata = results.get(0).getMetadata();
 			assertThat(normalizeValue(metadata.get("country"))).isEqualTo("BG");
-			assertThat(normalizeValue(metadata.get("year"))).isEqualTo("2023");
+			// the values are converted into Double
+			assertThat(normalizeValue(metadata.get("year"))).isEqualTo("2023.0");
```

Comment:
The metadata for number values get converted into double to avoid data loss. But, there can be a better way to handle.